packages/components/pie-button/test/component/pie-button.spec.ts

```diff
@@ -272,422 +156,228 @@ test.describe('Form Actions', () => {
-
-                    const span = document.createElement('span');
-                    span.id = 'formSubmittedFlag';
-                    document.body.appendChild(span);
-                });
-            });
+            const formIntegrationPage = new FormIntegrationPage(page, 'button--form-integration');
+            await formIntegrationPage.load();
 
             // Fill out the form
-            await page.fill('#username', 'testUser');
-            await page.fill('#password', 'testPassword');
+            await formIntegrationPage.fillForm(formInput);
 
             // Act
-            // Press Tab until we focus the reset button
-            await page.keyboard.press('Tab');
             await page.keyboard.press('Tab');
-
-            // Press Enter on the pie-button with type reset
-            await page.press('#resetPieButton', 'Enter');
+            await page.keyboard.press('Enter');
 
             // Assert
-            const formSubmittedFlagExists = Boolean(await page.$('#formSubmittedFlag'));
-            expect(formSubmittedFlagExists).toBe(false);
+            await expect(formIntegrationPage.formSubmittedFlag).toHaveCount(0);
         });
 
         test('should NOT submit the form when pressing Enter on a native non-submit button', async ({ page }) => {
             // Arrange
-            await page.evaluate(() => {
-                const formHTML = `
-                    <form id="testForm" action="/foo" method="POST">
-                        <input type="text" id="username" name="username" required>
-                        <input type="password" id="password" name="password" required>
-                        <pie-button type="submit">Submit</pie-button>
-                        <button id="resetNativeButton" type="reset">Reset</button>
-                    </form>
-                `;
-                document.body.innerHTML = formHTML;
-
-                const form = document.querySelector('#testForm') as HTMLFormElement;
-
-                form.addEventListener('submit', (e) => {
-                    e.preventDefault();
-
-                    const span = document.createElement('span');
-                    span.id = 'formSubmittedFlag';
-                    document.body.appendChild(span);
-                });
-            });
+            const formIntegrationPage = new FormIntegrationPage(page, 'button--form-integration');
+            await formIntegrationPage.load({ showNativeResetButton: true });
 
-            // Fill out the form
-            await page.fill('#username', 'testUser');
-            await page.fill('#password', 'testPassword');
+            await formIntegrationPage.fillForm(formInput);
 
             // Act
             // Press Enter on the native button with type reset
-            await page.focus('#resetNativeButton');
-            await page.press('#resetNativeButton', 'Enter');
+            await formIntegrationPage.focusButton('native-reset').then((locator) => locator.press('Enter'));
```

Comment:
I think mixing `await` and `.then` is a little confusing to read.

Would something like:

```
await formIntegrationPage.focusButton('native-reset');
await formIntegrationpage.clickButton('native-reset'); // Whatever the correct method is here
```

Be more consistent?

Reply:
I have no idea why I wrote this, will try changing it back 😂